test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1

```diff
@@ -246,6 +246,178 @@ Describe "Basic FileSystem Provider Tests" -Tags "CI" {
     }
 }
 
+Describe "Copy-Item can avoid copying an item onto itself" -Tags "CI" {
+    BeforeAll {
+        # In Windows, only Administrators can create symbolic links.
+        # In Unix, anyone can.
+        function CanMakeSymlink
+        {
+            if ($IsWindows)
+            {
+                $winId = [System.Security.Principal.WindowsIdentity]::GetCurrent()
+                $principal = New-Object System.Security.Principal.WindowsPrincipal($WinId)
+                $admin = [System.Security.Principal.WindowsBuiltInRole]::Administrator
+
+                return $principal.IsInRole($admin)
+            }
+            else
+            {
+                return $true
+            }
+        }
+
+        # For now, we'll assume the tests are running the platform's
+        # native filesystem, in its default mode
+        $isCaseSensitive = $IsLinux
+        $canSymlink = CanMakeSymlink
+        $canDirSymLink = $IsWindows -And (CanMakeSymlink)
+        $canJunction = $IsWindows
+
+        # The name of the key in an exception's Data dictionary when an
+        # attempt is made to copy an item onto itself.
+        $selfCopyKey = "SelfCopy"
+
+        # Names of files, directories, and links we'll be using to test
+        $subDir = "$TestDrive/sub"
+        $otherSubDir = "$TestDrive/other-sub"
+        $otherFile = "$otherSubDir/other-file"
+        $otherFileName = "other-file"
+        $otherFile = "$otherSubDir/$otherFileName"
+        $symToOther = "$subDir/sym-to-other"
+        $secondSymToOther = "$subDir/another-sym-to-other"
+        $symToSym = "$subDir/sym-to-sym-to-other"
+        $symToOtherFile = "$subDir/sym-to-other-file"
+        $hardToOtherFile = "$subDir/hard-to-other-file"
+        $symdToOther = "$subDir/symd-to-other"
+        $junctionToOther = "$subDir/junction-to-other"
+
+        # Set up our directories and links
+        New-Item -ItemType Directory $subDir >$null
+        New-Item -ItemType Directory $otherSubDir >$null
+        New-Item -ItemType File $otherFile -Value "some text" >$null
+        if ($canSymlink)
+        {
+            New-Item -ItemType SymbolicLink $symToOther -Value $otherSubDir >$null
+            New-Item -ItemType SymbolicLink $secondSymToOther -Value $otherSubDir >$null
+            New-Item -ItemType SymbolicLink $symToSym -Value $symToOther >$null
+            New-Item -ItemType SymbolicLink $symToOtherFile -Value $otherFile >$null
+        }
+        New-Item -ItemType HardLink $hardToOtherFile -Value $otherFile >$null
+
+        if ($canJunction)
+        {
+            New-Item -ItemType Junction $junctionToOther -Value $otherSubDir >$null
+        }
+        if ($canDirSymLink)
+        {
+            New-Item -ItemType SymbolicLink $symdToOther -Value $otherSubDir >$null
+        }
+
+        # Test the ability to avoid an item copying onto itself
+        function TestSelfCopy($testCase)
+        {
+            It "$($testCase.Name)" -Skip:($testCase.SkipIf) {
```

Comment:
With moving to `'RequireAdminOnWindows'` I believe we can use `It "" -TestCase $testCases` directly without `TestSelfCopy`